plugins/module_utils/utils.py

```diff
@@ -264,6 +299,205 @@ def op(operation, path, value):
+                                if dict1 not in merged_list:
+                                    merged_list.append(dict1)
+                                    matched_list.append(dict1)
+                            else:
+                                # They have the same value types, but do not match
+                                merged_dict = recursive_merge(merged_dict, dict2)
+                                matched_list.append(dict1)
+                                matched_list.append(dict2)
+                                if merged_dict not in merged_list:
+                                    merged_list.append(merged_dict)
+                                    matched_list.append(merged_dict)
+                    else:
+                        # They do not have anything in common, just add both to merge_list and matched_list
+                        if dict1 not in merged_list and dict1 not in matched_list:
+                            merged_list.append(dict1)
+                            matched_list.append(dict1)
+                        if dict2 not in merged_list and dict2 not in matched_list:
+                            # Add dict2 if already not present in merged_list
+                            merged_list.append(dict2)
+                            matched_list.append(dict2)
+            else:
+                if dict2 not in merged_list:
+                    merged_list.append(dict2)
+
+    return merged_list
+
+
+def merge_list_of_dicts(list1: List, list2) -> List:
+    # Handle when list of dicts (different than Tags and different key names) need to be merged
+    merged_list = list1.copy()
+
+    for dict2 in list2:
+        matching_indices = [
+            i for i, item1 in enumerate(merged_list) if any(k in item1 for k in dict2)
+        ]
+
+        if matching_indices:
+            for index in matching_indices:
+                if isinstance(merged_list, list):
+                    dict1 = merged_list[index]
+                    if isinstance(dict2, dict) and isinstance(dict1, dict):
+                        for key, value in dict2.items():
+                            value_dict = safe_json(value)
+                            if (
+                                dict1.get(key, {})
+                                and isinstance(value_dict, dict)
+                                and isinstance(dict1[key], dict)
+                            ):
+                                dict1[key] = recursive_merge(dict1[key], value_dict)
+                            elif (
+                                dict1.get(key, [])
+                                and isinstance(value_dict, list)
+                                and isinstance(dict1[key], list)
+                            ):
+                                # Easy merge
+                                if all(
+                                    isinstance(item, (str, int, float))
```

Comment:
Do we care about `bool`? A better type comparison here would probably be to use `collections.abc.Hashable` because that's what we really care about.